app/main.py

```diff
@@ -1,2 +1,100 @@
-class Dictionary:
-    pass
+from typing import Any, Iterable, Optional, Hashable
+
+
+class Dictionary:
+
+    def __init__(
+            self,
+            capacity: int = 8,
+            load_factor: float = 0.75
+    ) -> None:
+        self.length: int = 0
+        self.capacity: int = capacity
+        self.load_factor: float = load_factor
+        self.hash_table: list = [None] * capacity
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.get_index(key)
+        if not self.hash_table[index]:
+            self.hash_table[index] = []
+        bucket = self.hash_table[index]
+        for i, (keys, values) in enumerate(bucket):
+            if keys == key:
+                bucket[i] = (key, value)
+                return
+        bucket.append((key, value))

```

Comment:
don't use `append` 


now you store `tuple` inside `list`
```
[None, None, [(10, 2)], [(3, 1)], None, None, None, None]
```

Just store key and value inside `tuple` or `list`